atoma-auth/src/auth.rs

```diff
@@ -496,6 +542,72 @@ impl Auth {
         Ok(())
     }
 
+    async fn get_zk_address(zk_login_signature: &str, tx_digest: &str) -> Result<String> {
+        let user_signature = UserSignature::from_base64(zk_login_signature)?;
+        if let UserSignature::ZkLogin(zk_login_authenticator) = user_signature {
+            // The message is the transaction digest that we claim is ours
+            let intent_msg = IntentMessage::new(
+                Intent::personal_message(),
```

Comment:
I am confused by this

Reply:
Refactored into new function.
The original TS code where this message is created https://github.com/MystenLabs/ts-sdks/blob/main/packages/typescript/src/cryptography/intent.ts